src/script/ismine.cpp

```diff
@@ -49,7 +50,7 @@ static isminetype IsMineInner(const CKeyStore& keystore, const CScript& scriptPu
         break;
     case TX_PUBKEY:
         keyID = CPubKey(vSolutions[0]).GetID();
-        if (sigversion != IsMineSigVersion::BASE && vSolutions[0].size() != 33) {
+        if (sigversion != IsMineSigVersion::TOP && sigversion != IsMineSigVersion::P2SH && vSolutions[0].size() != 33) {
```

Comment:
commit: Track difference between scriptPubKey and P2SH execution in IsMine

I think a function `bool SigVersionPermitsUncompressed(IsMineSigVersion)` would improve clarity.

Reply:
Yes, that's much more readable. Done.